masu/database/report_stats_db_accessor.py

```diff
@@ -0,0 +1,215 @@
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Downloader for cost usage reports."""
+
+from datetime import datetime
+
+from masu.database.koku_database_access import KokuDBAccess
+
+
+class ReportStatsDBAccessor(KokuDBAccess):
+    """Class to interact with the koku database for CUR processing statistics."""
+
+    def __init__(self, report_name, provider_id=None, schema='public'):
+        """
+        Establish CUR statistics database connection.
+
+        Args:
+            report_name    (String) CUR report file name
+            provider_id    (String) the database id of the provider
+            schema         (String) database schema (i.e. public or customer tenant value)
+        """
+        super().__init__(schema)
+        self._provider_id = provider_id
+        self._report_name = report_name
+        self._costentrystatus = self.get_base().classes.reporting_common_costusagereportstatus
+
+        if self.does_db_entry_exist() is False:
+            update_fields = {}
+            update_fields['provider_id'] = self._provider_id
+            update_fields['report_name'] = self._report_name
+            update_fields['cursor_position'] = 0
+            self.add(update_fields)
+
+    def _get_db_obj_query(self):
+        """
+        Return the sqlachemy query for the report stats object.
+
+        Args:
+            None
+        Returns:
+            (sqlalchemy.orm.query.Query): "SELECT public.api_customer.group_ptr_id ..."
+        """
+        obj = self._session.query(self._costentrystatus).filter_by(report_name=self._report_name)
+        return obj
+
+    def commit(self):
+        """
+        Commit pending database changes.
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        self._session.commit()
+
+    def get_cursor_position(self):
+        """
+        Return current cursor position for processing CUR.
+
+        Args:
+            None
+        Returns:
+            (Integer): last byte offset for the given CUR file.
+        """
+        obj = self._get_db_obj_query().first()
+        return obj.cursor_position
+
+    def set_cursor_position(self, new_position):
+        """
+        Save current cursor position for processing CUR.
+
+        Args:
+            (Integer): last byte offset for the given CUR file.
+        Returns:
+            None
+
+        """
+        obj = self._get_db_obj_query()
+        new_atribute = {'cursor_position': new_position}
+        obj.update(new_atribute)
```

Comment:
Looking at http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.update see the warning. We are losing the benefit of the ORM by using `query.update()`

If we go directly with something like 

```
obj = self._get_db_obj_query().first()
obj.cursor_position = new_position
```

That puts the session in a "dirty" state, and the session tracks changes for us, and we're back in the ORM's good graces. 

Also should this just call the update method like in log_last_started_datetime below?

Reply:
Yes agree with this.  I can try to make this change here just so we know it works and then adopt this pattern across the other accessors (if there are anyway).

As for calling update internally.  Yeah its inconsistent now.  I'll experiment with this and try to make the change to move away from using update.